src/matrix_comps.jl

```diff
@@ -653,7 +653,25 @@ function predictor(sys::ST, R1, R2) where ST <: AbstractStateSpace
     predictor(sys, K)
 end
 
-function ControlSystems.predictor(sys, K::AbstractMatrix)
+function predictor(sys, K::AbstractMatrix)
     A,B,C,D = ssdata(sys)
     ss(A-K*C, [B K], C, [D zeros(size(D,1), size(K, 2))], sys.timeevol)
-end
\ No newline at end of file
+end
+
+"""
+    cont = controller(sys, L::AbstractMatrix, K::AbstractMatrix)
+
+Return the controller `cont` that is given by
+`ss(A - B*L - K*C + K*D*L, K, L, 0)`
+
+Such that `feedback(sys, cont)` produces a closed-loop system with eigenvalues given by `A-KC` and `A-BL`.
+
+# Arguments:
+- `sys`: Model of system
+- `L`: State-feedback gain `u = -Lx`
+- `K`: Observer gain
+"""
+function controller(sys, L::AbstractMatrix, K::AbstractMatrix)
+    A,B,C,D = ssdata(sys)
+    ss(A - B*L - K*C + K*D*L, K, L, 0, sys.timeevol)
```

Comment:
Wouldn't this correspond to the output of the controller being `-Lx`?
```suggestion
    ss(A - B*L - K*C + K*D*L, K, -L, 0, sys.timeevol)
```

Reply:
Ohh, you want the output to be `Lx` so that you can use negative feedback. Seems reasonable, but could maybe confuse more people than me. Maybe just state very clearly that the output of the controller is `Lx` and not `-Lx`.

Reply:
Mm, yeah I guess this is alright.

But what about the case when one wants the controller to treat the measurement and reference signals differently. I would say that this convention is still acceptable, although it would feel weird to add the measurement signal to the reference signal, instead of subtracting it.

Reply:
When you introduce reference signals as well it can get very messy. In RobustAndOptimalControl.jl I'm experimenting with using a type `ExtendedStateSpace` which is similar to, and should probably be merged with `PartitionedStateSpace`, such that one can defined the controller with references as
```julia
function extended_controller(K)
    nx,nu,ny = K.nx, K.nu, K.ny
    A,B,C,D = ssdata(K)
    ss(A, B, -B, zeros(0,nx), C, D21=D, D22=-D)
end
```
i.e., it has two inputs, `r` and `y`. I haven't landed on how to represent everything yet though, I have also experimented with systems with named signals, ideally this should be used as well to not rely on indices everywhere.

Reply:
Sounds good, not sure either what makes the most sense, so good that you are thinking about it. Yes, it seems more like a `PartitionedStateSpace`. We could change `PartitionedStateSpace`, I don't think we really need to explicitly use such a type for the delay systems.

Perhaps it won't even be needed here either with appropriate name of the signals.

Note: In this case one would have to use positive feedback?